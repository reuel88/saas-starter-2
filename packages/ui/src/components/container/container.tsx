import {
  ComponentPropsWithoutRef,
  ElementType,
  ForwardedRef,
  forwardRef,
} from "react";
import { cn } from "@repo/utilities";
import { BoxProps } from "../../types/box";

export const Container = forwardRef(
  <T extends ElementType = "div">(
    {
      as,
      children,
      className = "",
      ...rest
    }: BoxProps<T> & Omit<ComponentPropsWithoutRef<T>, keyof BoxProps<T>>,
    ref: ForwardedRef<any>,
  ) => {
    const Component = as || "div";

    return (
      <Component
        ref={ref}
        className={cn("mx-auto max-w-7xl px-6 lg:px-8", className)}
        {...rest}
      >
        {children}
      </Component>
    );
  },
);
