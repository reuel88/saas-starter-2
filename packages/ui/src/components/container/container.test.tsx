import { render, screen } from "@testing-library/react";
import { Container } from "./container";

describe("Container", () => {
  it("should render a div container with children when no 'as' prop is passed", () => {
    // Arrange

    // Act
    const { baseElement } = render(
      <Container data-testid="container">Test</Container>,
    );
    const containerElement = screen.getByTestId("container");

    // Assert
    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(containerElement).toHaveAttribute(
      "class",
      "mx-auto max-w-7xl px-6 lg:px-8",
    );
    expect(baseElement).toMatchSnapshot();
  });
});
