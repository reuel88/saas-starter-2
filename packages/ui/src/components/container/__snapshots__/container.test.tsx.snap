// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > should render a div container with children when no 'as' prop is passed 1`] = `
<body>
  <div>
    <div
      class="mx-auto max-w-7xl px-6 lg:px-8"
      data-testid="container"
    >
      Test
    </div>
  </div>
</body>
`;
